src/nvim/api/vimscript.c

```diff
@@ -849,15 +848,24 @@ Dictionary nvim_parse_cmd(String str, Dictionary opts, Error *err)
     PUT(result, "cmd", CSTR_TO_OBJ((char *)get_command_name(NULL, ea.cmdidx)));
   }
 
-  PUT(result, "range", INTEGER_OBJ(ea.addr_count));
-  PUT(result, "line1", INTEGER_OBJ((ea.argt & EX_RANGE) ? ea.line1 : -1));
-  PUT(result, "line2", INTEGER_OBJ((ea.argt & EX_RANGE) ? ea.line2 : -1));
+  if ((ea.argt & EX_RANGE) && !(ea.argt & EX_COUNT) && ea.addr_count > 0) {
+    Array range = ARRAY_DICT_INIT;
+    if (ea.addr_count > 1) {
+      ADD(range, INTEGER_OBJ(ea.line1));
+    }
+    ADD(range, INTEGER_OBJ(ea.line2));
+    PUT(result, "range", ARRAY_OBJ(range));;
+  } else {
+    PUT(result, "range", ARRAY_OBJ(ARRAY_DICT_INIT));
+  }
 
   if (ea.argt & EX_COUNT) {
-    if (ea.addr_count > 0 || cmd == NULL) {
+    if (ea.addr_count > 0) {
       PUT(result, "count", INTEGER_OBJ(ea.line2));
```

Comment:
should this be `ea.addr_count` ?

Reply:
Are you asking if `ea.addr_count` is the variable that needs to be checked? Then yes.